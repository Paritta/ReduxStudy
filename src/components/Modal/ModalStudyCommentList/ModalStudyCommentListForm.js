import React from "react";
import styled from "styled-components";
import { Field, reduxForm } from "redux-form";
import oc from "open-color";

const Wrapper = styled.div`
    position: absolute;
    
    bottom: 0;
    left: 0;
    
    width: 100%;
`;

const StyledField = styled(Field)`
    background-color: ${oc.gray[0]};
    width: 100%;
    height: 40px;
    font-size: 15px;
    border: 1px solid white;
    margin: 0;
    padding: 0 0 0 20px;
    border-radius: 0;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    padding: 0.7em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 0;
    
    transition: all 0.3s easeo
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
    
    flex: 1;
`;

const Container = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
`;

const ButtonWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: row;
`;

const Form = styled.form`
    width: 100%;
`;

export let ModalStudyCommentListForm = ({ handleSubmit,hideModal }) => {
    return (
        <Wrapper>
            <Form onSubmit={ handleSubmit }>
                <Container>
                    <StyledField name="comment" component="input" type="text" required placeholder="댓글을 입력하세요 ..."/>
                    <ButtonWrapper>
                        <Button type="submit">전송</Button>
                        <Button onClick={() => hideModal()}>나가기</Button>
                    </ButtonWrapper>
                </Container>
            </Form>
        </Wrapper>
    )
};

ModalStudyCommentListForm = reduxForm({
    form: "ModalSignIn"
})(ModalStudyCommentListForm);

export default ModalStudyCommentListForm;
