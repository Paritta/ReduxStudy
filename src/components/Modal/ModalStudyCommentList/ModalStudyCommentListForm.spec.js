import React from "react";
import { ModalStudyCommentListForm } from "./ModalStudyCommentListForm";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

it("Layout render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Provider store={store}>
            <ModalStudyCommentListForm />
        </Provider> 
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
