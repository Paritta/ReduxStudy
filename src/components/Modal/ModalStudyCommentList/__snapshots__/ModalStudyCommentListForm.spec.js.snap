// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render correctly 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c1 {
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c3 {
  background-color: #f8f9fa;
  width: 100%;
  height: 40px;
  font-size: 15px;
  border: 1px solid white;
  margin: 0;
  padding: 0 0 0 20px;
  border-radius: 0;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  padding: 0.7em 1em;
  border: 2px solid palevioletred;
  border-radius: 0;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5:hover {
  background: #f783ac;
  border: 2px solid #f783ac;
}

<div
  className="c0"
>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        name="comment"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="댓글을 입력하세요 ..."
        required={true}
        type="text"
        value=""
      />
      <div
        className="c4"
      >
        <button
          className="c5"
          type="submit"
        >
          전송
        </button>
        <button
          className="c5"
          onClick={[Function]}
        >
          나가기
        </button>
      </div>
    </div>
  </form>
</div>
`;
